### PR TITLE
ci: Add an eve step to retrieve, install and test terraform

### PR DESCRIPTION
**Pull request by Ebaneck**
**Component**:

'ci', 'flakiness'

**Context**: 

See #1845

**Summary**:

- Ensure the terraform binary is available when required.

**Acceptance criteria**: 

- Green build.
- Successful install of terraform and all dependencies required to set it up.
---

Closes: #1845

### CODE REVIEW

File: eve/main.yml [hunk 1]
```diff
@@ -234,6 +234,37 @@ stages:
...
+              exit 1
+            fi
+          haltOnFailure: true
+      - ShellCommand:
+          name: Download and install terraform
+          command: >
+            TERRAFORM_VERSION=0.12.3
+            tf_url="https://releases.hashicorp.com/terraform/";
+            tf_suburl="${TERRAFORM_VERSION}/";
+            tf_file="terraform_${TERRAFORM_VERSION}_linux_amd64.zip";
```
Comment (tcarmet) [inline eve/main.yml]: optional: If you want, you can use the `env` option on the `ShellCommand` step to set those variables.

Like the following:
```yaml
ShellCommand:
    name: doing stuff
    command: do stuff
    env:
      TERRAFORM_VERSION: '0.12.3'
...
```

**Review by charlesprost — DISMISSED** [review]

**Review by charlesprost — DISMISSED** [review]

**Review by charlesprost — APPROVED** [review]

**Comment by charlesprost** [comment]
Rebuilding.

**Pull request merged**